src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHash.cs

```diff
@@ -0,0 +1,318 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class PreferHashDataOverComputeHashAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string CA1847 = nameof(CA1847);
+        internal const string TargetHashTypeName = nameof(TargetHashTypeName);
```

Comment:
Probably rename this to say something like `TargetHashTypeDiagnosticPropertyKey` to make it more explicit that is a a key for an entry added to diagnostic's property bag.